hdfs/src/object_store/hdfs.rs

```diff
@@ -229,7 +229,8 @@ impl ObjectStore for HadoopFileSystem {
         &self,
         prefix: Option<&Path>,
     ) -> object_store::Result<BoxStream<'_, object_store::Result<ObjectMeta>>> {
-        let prefix = prefix.expect("Prefix for hdfs should not be None");
+        let default_path = Path::from(self.get_path_root());
+        let prefix = prefix.unwrap_or(&default_path);
```

Comment:
To align with the semantics of LocalFileSystem's 'list' function, which provides a default root when a prefix is not given. Please refer to https://github.com/apache/arrow-rs/blob/master/object_store/src/local.rs#L437-L440.